packages/gemini-runner/src/adapters.ts

```diff
@@ -0,0 +1,149 @@
+import type { SDKMessage, SDKUserMessage } from "cyrus-core";
+import type { GeminiMessageEvent, GeminiStreamEvent } from "./types.js";
+
+/**
+ * Convert a Gemini stream event to Claude SDK message format
+ *
+ * This adapter maps Gemini CLI's streaming events to the Claude SDK's SDKMessage
+ * format, allowing GeminiRunner to implement the IAgentRunner interface.
+ *
+ * @param event - Gemini CLI stream event
+ * @param sessionId - Current session ID (may be null initially)
+ * @returns SDKMessage or null if event type doesn't map to a message
+ */
+export function geminiEventToSDKMessage(
+	event: GeminiStreamEvent,
+	sessionId: string | null,
+): SDKMessage | null {
+	switch (event.type) {
+		case "message": {
+			const messageEvent = event as GeminiMessageEvent;
+			if (messageEvent.role === "user") {
+				return {
+					type: "user",
+					message: {
+						role: "user",
+						content: messageEvent.content,
+					},
+					parent_tool_use_id: null,
+					session_id: sessionId || "pending",
+				} satisfies SDKUserMessage;
```

Comment:
instead of writing this way, and assuming you've presented the right info, how about defining a value which is a SDKUserMessage and defining anobject which satisfies it then we will get the benefits of type checking. Then return it.